### DISCUSSION TITLE
streamlink causes powershell script to hang

### DISCUSSION THREAD

**Discussion by COLEplusTEN** [streamlink/streamlink#3730] [Q&A] [answered]
Hello all,

I have a little powershell script to open twitch urls in vlc using streamlink. I'm using a do until loop and I want to keep prompting for user input but the streamlink command stops all processing.

Is there a way to get around this?

```ps1
$Title = "twitch_vlc"
$host.UI.RawUI.WindowTitle = $Title
$msg = 'Enter channel (along with any optional VLC arguments) [enter exit to close]'
do {
    $response = (Read-Host -Prompt $msg).Split(" ")
    if ($response -ne 'exit') {
        $url = "twitch.tv/" + $response[0]
        if($response.Length -le 1) {
            $arg = ""
        }
        else {
            $arg = $response[1..($response.Length-1)]
        }
        $playerpath = "--player=C:\Program Files (x86)\VideoLAN\VLC\vlc --aspect-ratio 180:55 " + $arg
        $activevlc = Get-Process vlc -ErrorAction SilentlyContinue
        if ($null -ne $activevlc) {
            Stop-Process -Name vlc
        }
        streamlink $playerpath $url best -Q --player-no-close --twitch-disable-ads --player-passthrough hls --stream-segment-attempts 1000 --stream-segment-timeout 10 --hls-segment-attempts 1000 --hls-segment-timeout 10
    }
    if ($response -eq 'exit') {
        Stop-Process -Name vlc
        exit
    }
} until ($response -eq 'exit')
```

**Comment by bastimeyer** [comment, marked as answer]
> but the streamlink command stops all processing

Yes, because the shell is waiting for the process to finish, which is default behavior in pretty much every command line shell. Your question is unrelated to streamlink as you could use any other process here instead.

https://stackoverflow.com/questions/185575/powershell-equivalent-of-bash-ampersand-for-forking-running-background-proce/53892094

**Reply by COLEplusTEN** [reply]
I appreciate the link! I was searching for hours trying to figure this out. I had a feeling it wasn't a streamlink issue, I guess I just didn't know how to properly word my search. Thank you for your quick reply and all the work you do/have done.
